### PR TITLE
(FACT-2107) Created fips_enabled resolver for fedora.

### PR DESCRIPTION
**Pull request by gigicampean**

### CODE REVIEW

File: lib/resolvers/fips_enabled_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+              result ||= @fact_list[fact_name]
+              subscribe_to_manager
+              result || read_fips_file(fact_name)
+            end
+          end
+
+          def read_fips_file(fact_name)
+            file_output = File.read('/proc/sys/crypto/fips_enabled')
+            @fact_list[:fips_enabled] = 'false'
+            @fact_list[:fips_enabled] = 'true' if file_output.to_i == 1
```
Comment (IrimieBogdan) [inline lib/resolvers/fips_enabled_resolver.rb]: I would replace this code snippet with 
` @fact_list[:fips_enabled] = file_output.to_i == 1 ? true : false`

Comment (sebastian-miclea) [reply lib/resolvers/fips_enabled_resolver.rb]: I would replace it with
`@fact_list[:fips_enabled] = file_output == '1'`

File: spec/facter/resolvers/fips_enabled_resolver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+describe 'FipsEnabledResolver' do
+  let(:fips_enabled) { 'false' }
+
+  before do
+    allow(File).to receive(:read)
+      .with('/proc/sys/crypto/fips_enabled')
+      .and_return(load_fixture('fips_enabled').read)
```
Comment (oanatmaria) [inline spec/facter/resolvers/fips_enabled_resolver_spec.rb]: there is no need for a fixture file in this case. I would make a variable which you can set/unset and make a test scenario when is not fips_enabled also

File: spec/facter/resolvers/fips_enabled_resolver_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+describe 'FipsEnabledResolver' do
+  describe '#resolve' do
+    context 'when fips is not enabled' do
+      let(:fips_enabled) { false }
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/fips_enabled_resolver_spec.rb]: I think we could drop this var and just add the value to the expect statement.

**Review by sebastian-miclea — APPROVED** [review]

**Pull request merged**